src/processTargets/modifiers/ItemStage/ItemStage.ts

```diff
@@ -0,0 +1,169 @@
+import { Range, TextEditor } from "vscode";
+import { NoContainingScopeError } from "../../../errors";
+import { Target } from "../../../typings/target.types";
+import {
+  ContainingScopeModifier,
+  EveryScopeModifier,
+  SimpleScopeTypeType,
+} from "../../../typings/targetDescriptor.types";
+import { ProcessedTargetsContext } from "../../../typings/Types";
+import { getInsertionDelimiter } from "../../../util/nodeSelectors";
+import { rangeLength } from "../../../util/rangeUtils";
+import { ModifierStage } from "../../PipelineStages.types";
+import ScopeTypeTarget from "../../targets/ScopeTypeTarget";
+import ContainingSyntaxScopeStage, {
+  SimpleContainingScopeModifier,
+} from "../scopeTypeStages/ContainingSyntaxScopeStage";
+import { getIterationScope } from "./getIterationScope";
+import { tokenizeRange } from "./tokenizeRange";
+
+export default class ItemStage implements ModifierStage {
+  constructor(private modifier: ContainingScopeModifier | EveryScopeModifier) {}
+
+  run(context: ProcessedTargetsContext, target: Target): Target[] {
+    // First try the language specific implementation of item
+    try {
+      return new ContainingSyntaxScopeStage(
+        <SimpleContainingScopeModifier>this.modifier
+      ).run(context, target);
+    } catch (_error) {}
+
+    // Then try the textual implementation
+    if (this.modifier.type === "everyScope") {
+      return this.getEveryTarget(context, target);
+    }
+    return [this.getSingleTarget(context, target)];
+  }
+
+  private getEveryTarget(context: ProcessedTargetsContext, target: Target) {
+    const itemInfos = getItemInfosForIterationScope(context, target);
+
+    // If weak expand to all items in iteration scope
+    const filteredItemInfos = target.isWeak
+      ? itemInfos
+      : filterItemInfos(target, itemInfos).map((e) => e.itemInfo);
+
+    if (filteredItemInfos.length === 0) {
+      throw new NoContainingScopeError(this.modifier.scopeType.type);
+    }
+
+    return filteredItemInfos.map((itemInfo) =>
+      this.itemInfoToTarget(target, itemInfo)
+    );
+  }
+
+  private getSingleTarget(context: ProcessedTargetsContext, target: Target) {
+    const itemInfos = getItemInfosForIterationScope(context, target);
+
+    const itemInfoWithIntersections = filterItemInfos(target, itemInfos);
+
+    if (itemInfoWithIntersections.length === 0) {
+      throw new NoContainingScopeError(this.modifier.scopeType.type);
+    }
+
+    itemInfoWithIntersections.sort(
+      (a, b) =>
+        rangeLength(target.editor, b.intersection!) -
+        rangeLength(target.editor, a.intersection!)
+    );
+
+    return this.itemInfoToTarget(target, itemInfoWithIntersections[0].itemInfo);
```

Comment:
Should just make a range from first to last item.  See test I added.  That behaviour is consistent with "line", and desired behaviour in https://github.com/cursorless-dev/cursorless/issues/484

Reply:
To day I don't think we want `take item comma` to select both items. This would break with the behavior of arguments.

Reply:
Okay then let's do that